### ISSUE TITLE
OpenAPI Import "Null Pointer Exception" in weekly

### ISSUE THREAD

**Issue by globeone** [zaproxy/zaproxy#6205]
**Describe the bug**
When importing a valid OpenAPI JSON doc (validated with external tools in VS Code) from URL or File ZAP_D_2020-09-29 Weekly throws a java "Null pointer exception" and fails to import.

When importing the exact same OpenAPI Swagger JSON doc into ZAP 2.9.0 everything works as expected and the endpoints start to get scanned.

**To Reproduce**
Prereq: Download and run the ZAP_D_2020-09-29 Weekly version
Have a valid OpenAPI Swagger JSON doc
1. Go to 'Import an OpenAPI definition from a {file,URL}' (doesn't matter which one)
2. Enter on the URL and the Start URL or import the JSON file from disk
4. See error: 
WARN  OpenAPIV3Parser - Exception while resolving:
java.lang.NullPointerException
WARN  ExtensionOpenApi - Failed to parse swagger defn null

**Expected behavior**
ZAP to import the OpenAPI Swagger doc and start parsing and scanning all the endpoints

**Screenshots**

**Software versions**
 - ZAP: ZAP_D-2020-09-29
 - Add-on: No additional addons used
 - OS: Windows 10 - 2004 (19041.508)
 - Java: Bellsoft Liberica
openjdk 14.0.2 2020-07-14
OpenJDK Runtime Environment (build 14.0.2+13)
OpenJDK 64-Bit Server VM (build 14.0.2+13, mixed mode, sharing)
 - Browser: [e.g. chrome 66, firefox 59, safari 11.1]

**Errors from the zap.log file**
```
2020-09-30 14:43:19,885 [ZAP-Import-OpenAPI-6] WARN  OpenAPIV3Parser - Exception while resolving:
java.lang.NullPointerException
	at io.swagger.v3.parser.util.ResolverFully.resolveFully(ResolverFully.java:60)
	at io.swagger.v3.parser.OpenAPIV3Parser.resolve(OpenAPIV3Parser.java:174)
	at io.swagger.v3.parser.OpenAPIV3Parser.readContents(OpenAPIV3Parser.java:153)
	at io.swagger.v3.parser.OpenAPIV3Parser.readContents(OpenAPIV3Parser.java:99)
	at io.swagger.parser.OpenAPIParser.readContents(OpenAPIParser.java:29)
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.parseV2(SwaggerConverter.java:236)
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.getOpenAPI(SwaggerConverter.java:201)
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.readOpenAPISpec(SwaggerConverter.java:171)
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.getRequestModels(SwaggerConverter.java:158)
	at org.zaproxy.zap.extension.openapi.ExtensionOpenApi$3.run(ExtensionOpenApi.java:236)
2020-09-30 14:43:19,886 [ZAP-Import-OpenAPI-6] WARN  OpenAPIV3Parser - Exception while resolving:
java.lang.NullPointerException
	at io.swagger.v3.parser.util.ResolverFully.resolveFully(ResolverFully.java:60)
	at io.swagger.v3.parser.OpenAPIV3Parser.resolve(OpenAPIV3Parser.java:174)
	at io.swagger.v3.parser.OpenAPIV3Parser.readContents(OpenAPIV3Parser.java:153)
	at io.swagger.v3.parser.OpenAPIV3Parser.readContents(OpenAPIV3Parser.java:99)
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.parseV2(SwaggerConverter.java:240)
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.getOpenAPI(SwaggerConverter.java:201)
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.readOpenAPISpec(SwaggerConverter.java:171)
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.getRequestModels(SwaggerConverter.java:158)
	at org.zaproxy.zap.extension.openapi.ExtensionOpenApi$3.run(ExtensionOpenApi.java:236)
2020-09-30 14:43:28,022 [ZAP-Import-OpenAPI-6] WARN  ExtensionOpenApi - Failed to parse swagger defn null
org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerException: Failed to parse swagger defn null
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.readOpenAPISpec(SwaggerConverter.java:174)
	at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.getRequestModels(SwaggerConverter.java:158)
	at org.zaproxy.zap.extension.openapi.ExtensionOpenApi$3.run(ExtensionOpenApi.java:236)
```

**Additional context**
Like said earlier the same JSON file imported into 2.9.0 stable works without any issues. So, there is regression going on here.

**Would you like to help fix this issue?**
Happy to test.

**Comment by thc202** [comment]
Could you provide a sample definition that reproduces the issue? The exception is happening in the Swagger library, which was updated recently.

**Comment by globeone** [comment]
Unfortunately, it's for an internal API that can't be shared. I tried to sanitize the Swagger Doc but was not successful. If you can give me specific areas of the Swagger doc that you suspect then I can possibly sanitize parts of the doc and share that. 
I was hoping to get some time this week to test with Swagger Petshop as the same null pointer exception should occur there as well.

**Comment by thc202** [comment]
It seems that there's a path parameter in an operation but that parameter is not actually defined.

**Comment by thc202** [comment]
The exception was already reported and fixed upstream, swagger-api/swagger-parser#1402. We'll keep an eye on the next release.

**Comment by globeone** [comment]
I'll test again with the same API in the next weekly version.
